libraries/core_libs/consensus/src/rewards/rewards_stats.cpp

```diff
@@ -75,9 +75,31 @@ std::vector<BlockStats> Stats::processStats(const PeriodData& current_blk, const
     return {};
   }
 
-  res.reserve(blocks_stats_.size());
-  std::transform(blocks_stats_.begin(), blocks_stats_.end(), std::back_inserter(res),
-                 [](auto& t) { return std::move(t.second); });
+  // Transform ordered (or unordered) blocks stats into vector
+  auto transformStatsToVector = [](auto&& blocks_stats) {
+    std::vector<BlockStats> stats_vec;
+    stats_vec.reserve(blocks_stats.size());
+
+    std::transform(blocks_stats.begin(), blocks_stats.end(), std::back_inserter(stats_vec),
+                   [](auto&& t) { return std::move(t.second); });
+    return stats_vec;
+  };
```

Comment:
I will simply delete whole move semantics from here. The objects we are trying to move cannot be moved anyway because  BlockStats contains only integers... so it is always copied